### PR TITLE
Add Java Local Tests to Proxy Nightly Integration Tests

### LINKED ISSUE #6084: Add `java` local invocation to test-proxy livetests
Very straightforward repro, grab one of the standalone executables.

```powershell
C:\Users\billwert\AppData\Local\Temp\test-proxy\Azure.Sdk.Tools.TestProxy.exe --storage-location . -- --urls http://localhost:5000
'--urls' was not matched. Did you mean one of the following?
--help
Unrecognized command or argument '--urls'.
Unrecognized command or argument 'http://localhost:5000'.
```

Aha, this is because the `--urls` additional argument is only honored by the `start` verb now. A default server start (verbless) is assumed to be default startup only.

This broke the Java folks local invocation, as @billwert notified me.

To avoid this going forward, I have retitled this issue from `-- --urls not being honored by standalone test proxy` to `add java local invocation to test-proxy livetests`.

### PR DESCRIPTION
**Pull request by scbedd**
CC @billwert @samvaity 

Ensuring that I can't accidentally break your use-case in the future!

Resolves #6084

### CODE REVIEW

File: tools/test-proxy/tests.yml [hunk 1]
```diff
@@ -216,4 +220,59 @@ stages:
...
+
+        - template: /eng/pipelines/templates/steps/use-python-version.yml
+          parameters:
+            versionSpec: '3.9'
+
+        - pwsh: |
+            git clone https://github.com/$(REPO) $(CLONE_LOCATION) --depth 1
+          displayName: Clone Repo
+
+        - pwsh: |
```
Comment (billwert) [inline tools/test-proxy/tests.yml:260]: You may consider skipping more stuff to make this faster:

```
-Dcodesnippet.skip -Drevapi.skip -Dspotbugs.skip -DskipTests -Djacoco.skip -DskipTestCompile
```

Of course for the actual `azure-core-test` don't skiptests or test compile.

**Review by billwert — APPROVED** [review]
left a suggestion, lgtm.

File: tools/test-proxy/tests.yml (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/tests.yml:260]: 👍 I'll do so. Thanks Bill!

**Pull request merged**